text/0000-trait-alias.md

```diff
@@ -0,0 +1,270 @@
+our backend:
+
+```rust
+pub trait FooBackend: Foo<Backend> {
+  // ...
+}
+
+fn use_foo<A>(_: A) where A: FooBackend {}
+```
+
+If you try to pass an object that implements `Foo<Backend>`, that won’t work, because it doesn’t
+implement `FooBackend`. However, we can make it work with the following universal `impl`:
+
+```rust
+impl<T> FooBackend for T where T: Foo<Backend> {}
+```
+
+With that, it’s now possible to pass an object that implements `Foo<Backend>` to a function
+expecting a `FooBackend`. However, what about impl blocks? What happens if we implement only
+`FooBackend`? Well, we cannot, because the trait explicitely states that we need to implement
+`Foo<Backend>`. We hit a problem here. The problem is that even though there’s a compatibility at
+the `trait bound` level between `Foo<Backend>` and `FooBackend`, there’s none at the `impl` level,
+so all we’re left with is implementing `Foo<Backend>` – that will also provide an implementation for
+`FooBackend` because of the universal implementation just above.
+
+## Second example: ergonomic collections and scrapping boilerplate
+
+Another example is associated types. Take the following [trait from tokio](https://docs.rs/tokio-service/0.1.0/tokio_service/trait.Service.html):
+
+```rust
+pub trait Service {
+  type Request;
+  type Response;
+  type Error;
+  type Future: Future<Item=Self::Response, Error=Self::Error>;
+  fn call(&self, req: Self::Request) -> Self::Future;
+}
+```
+
+It would be nice to be able to create a few aliases to remove boilerplate for very common
+combinations of associated types with `Service`.
+
+```rust
+Service<Request = http::Request, Response = http::Response, Error = http::Error>;
+```
+
+The trait above is a http service trait which only the associated type `Future` is left to be
+implemented. Such an alias would be very appealing because it would remove copying the whole
+`Service` trait into use sites – trait bounds, or even trait impls. Scrapping such an annoying
+boilerplate is a definitive plus to the language and might be one of the most interesting use case.
+
+# Detailed design
+[design]: #detailed-design
+
+## Syntax
+
+The syntax chosen to make a *trait alias* is:
+
+```rust
+trait TraitAlias = Trait;
+```
+
+Trait aliasing to combinations of traits is also provided with the standard `+` construct:
+
+```rust
+trait DebugDefault = Debug + Default;
+```
+
+Optionally, if needed, one can provide a `where` clause to express *bounds*:
+
+```rust
+trait DebugDefault = Debug where Self: Default; // same as the example above
+```
+
+Furthermore, it’s possible to use only the `where` clause by leaving the list of traits empty:
+
+```rust
+trait DebugDefault = where Self: Debug + Default;
+```
+
+Specifically, the grammar being added is, in informal notation:
+
+```
+ATTRIBUTE* VISIBILITY? trait IDENTIFIER(<GENERIC_PARAMS>)? = GENERIC_BOUNDS (where PREDICATES)?;
```

Comment:
I read the whole Syntax section, and failed to see explicit mention of support for the `Trait<Item=Type>` syntax.

I infer that you want to support this syntax from the Motivation section of the RFC, specifically from your second example with the `Service` trait that has some specific concrete types already instantiated for  some of the associated types.

(If I recall correctly, Rust's current `where` clauses cannot express type equality constraints, so this really is more than just sugar on top of what you've already specified here.)